algebra/group.lean

```diff
@@ -598,3 +598,49 @@ end is_group_anti_hom
 
 theorem inv_is_group_anti_hom [group α] : is_group_anti_hom (λ x : α, x⁻¹) :=
 ⟨mul_inv_rev⟩
+
+/-- Predicate for additive group homomorphism. -/
+def is_add_group_hom {α β : Type*} [add_group α] [add_group β] (f : α → β) : Prop :=
+@is_group_hom (multiplicative α) (multiplicative β) _ _ f
+
+attribute [class] is_add_group_hom
+
+namespace is_add_group_hom
+
+-- variables {α β : Type*} [add_group α] [add_group β] (f : α → β) [hf : is_add_group_hom f]
+
+theorem mk {α β : Type*} [add_group α] [add_group β] (f : α → β) [hf : is_add_group_hom f]
+(H : ∀ x y, f (x + y) = f x + f y) : is_add_group_hom f := ⟨H⟩
+
+theorem add {α β : Type*} [add_group α] [add_group β] (f : α → β) [hf : is_add_group_hom f]
+(x y) : f (x + y) = f x + f y :=
+@is_group_hom.mul (multiplicative α) (multiplicative β) _ _ f hf x y
+
+-- instance ring_hom_is_add_group_hom [ring α] [ring β] (f : α → β) [is_ring_hom f] : is_add_group_hom f :=
+-- ⟨λ _ _, is_ring_hom.map_add f⟩
+
+theorem zero {α β : Type*} [add_group α] [add_group β] (f : α → β) [hf : is_add_group_hom f] :
```

Comment:
Should this be a simp lemma?

Reply:
This can't be a `simp` lemma as the head of the left-hand side is just `f` which is a variable. The simplifier cannot work with such cases.